### PR TITLE
Add `Tooltip` component

### PR DESCRIPTION
**Pull request by GuillaumeRx**
This adds a new component `Tooltip` which can be used to display a tooltip on a text element or an image.

```tsx
<Tooltip content="foo">
  <Text>bar</Text>
</Tooltip>
```

This component can wrap any text element (`Text`, `Italic`, `Bold`, `Link`) and `Image` and accepts a string or a text element (`Text`, `Italic`, `Bold`, `Link`) to be displayed inside it via its `value` prop.

### CODE REVIEW

File: packages/snaps-sdk/src/jsx/components/Tooltip.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  | null;
+
+/**
+ * The props of the {@link Tooltip} component.
+ *
+ * @property children - The children of the box.
+ */
+export type TooltipProps = {
+  children: TooltipChildren;
+  value: TextElement | StandardFormattingElement | LinkElement | string;
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/components/Tooltip.ts:21]:
```suggestion
  label: TextElement | StandardFormattingElement | LinkElement | string;
```
Potentially a better name? WDYT?

File: packages/snaps-sdk/src/jsx/components/Tooltip.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  | TextElement
+  | StandardFormattingElement
+  | LinkElement
+  | ImageElement
+  | null;
+
+/**
+ * The props of the {@link Tooltip} component.
+ *
+ * @property children - The children of the box.
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/components/Tooltip.ts:17]: Please document the other prop as well

File: packages/snaps-sdk/src/jsx/components/Tooltip.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  value: TextElement | StandardFormattingElement | LinkElement | string;
+};
+
+const TYPE = 'Tooltip';
+
+/**
+ * A tooltip component, which is used to display text in a tooltip.
+ *
+ * @param props - The props of the component.
+ * @param props.children - The children of the tooltip.
```
Comment (FrederikBolding) [inline packages/snaps-sdk/src/jsx/components/Tooltip.ts:30]: Please document the other prop as well

File: packages/snaps-sdk/src/jsx/components/Tooltip.ts (hunk 1 above)
Comment (GuillaumeRx) [reply packages/snaps-sdk/src/jsx/components/Tooltip.ts:21]: I'm not sure `label` is the correct term to define the content of a Tooltip 🤔

Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/components/Tooltip.ts:21]: `content` could work too 🤔 

E.g. Chakra uses `label` and NextUI uses `content`. Just trying to pick something that devs might be used to

Comment (GuillaumeRx) [reply packages/snaps-sdk/src/jsx/components/Tooltip.ts:21]: I'm okay with content :)

File: packages/snaps-sdk/src/jsx/components/Tooltip.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * @param props.content - The text to display in the tooltip.
+ * @returns A tooltip element.
+ * @example
+ * <Tooltip content={<Text>foo</Text>}>
+ *   <Text>Hello world!</Text>
+ * </Tooltip>
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/components/Tooltip.ts:37]: To make it clear that it takes strings too:
```suggestion
 * @example
 * <Tooltip content="Tooltip text">
 *   <Text>Hello world!</Text>
 * </Tooltip>
 * @example
 * <Tooltip content={<Text>Text with <Bold>formatting</Bold></Text>}>
 *   <Text>Hello world!</Text>
 * </Tooltip>
```

File: packages/snaps-sdk/src/jsx/components/Tooltip.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ */
+export const Tooltip = createSnapComponent<TooltipProps, typeof TYPE>(TYPE);
+
+/**
+ * A box element.
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/components/Tooltip.ts:42]:
```suggestion
 * A tooltip element.
```

**Review by FrederikBolding — DISMISSED** [review]
LGTM, but please address @Mrtenz comments as well!

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**